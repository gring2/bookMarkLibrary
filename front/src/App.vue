<template>
  <div id="app">
    <Header ></Header>
    <router-view/>
  </div>
</template>


<script lang="ts">
  import { Component, Vue } from 'vue-property-decorator'
  import Header from '@/views/Header.vue' // @ is an alias to /src
  import User from '@/vo/User'
  import { mapState } from 'vuex'
@Component({
    components: {
      Header,
    },
  })

export default class App extends Vue {
}
</script>

<style lang="scss">
  @import '@/styles/init.scss'

  /*#app {*/
/*  font-family: 'Avenir', Helvetica, Arial, sans-serif;*/
/*  -webkit-font-smoothing: antialiased;*/
/*  -moz-osx-font-smoothing: grayscale;*/
/*  text-align: center;*/
/*  color: #2c3e50;*/
/*}*/
/*#nav {*/
/*  padding: 30px;*/
/*  a {*/
/*    font-weight: bold;*/
/*    color: #2c3e50;*/
/*    &.router-link-exact-active {*/
/*      color: #42b983;*/
/*    }*/
/*  }*/
/*}*/
</style>
