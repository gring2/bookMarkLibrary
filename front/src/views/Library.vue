<template>
  <section class="content">
    <div>
      <TagsPanel></TagsPanel>
      <BookmarksPanel></BookmarksPanel>
    </div>

  </section>
</template>

<script lang="ts">
  import { Component, Prop, Vue } from 'vue-property-decorator'
  import TagsPanel from '@/views/TagsPanel.vue'
  import BookmarksPanel from '@/views/BookmarksPanel.vue'
  import {bookmarkMod} from '@/stores/modules/bookmark'

  @Component({
    components: {
      TagsPanel,
      BookmarksPanel
    },
  })
  export default class Library extends Vue {
    created(){
      bookmarkMod.GET_BOOKMARKS()
    }

  }
</script>

<!--<style scoped lang="scss">-->
<!--  @import "@/styles/header.scss"-->
<!--</style>-->

<style scoped lang="scss">
  @import "@/styles/library/urls.scss";
</style>

