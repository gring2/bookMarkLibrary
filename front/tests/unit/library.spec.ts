import { createLocalVue, mount } from '@vue/test-utils'
import Library from '@/views/Library.vue'
import VueRouter from 'vue-router'
import App from '@/App.vue'


describe('Library.vue', () => {
  it('renders a component via routing', () => {
    const localVue = createLocalVue()
    localVue.use(VueRouter)

    const router = new VueRouter({
      mode: 'history',
          base: process.env.BASE_URL,
          routes: [
            {
              path: '/bookmarks',
              name: 'bookmarks',
              // route level code-splitting
              // this generates a separate chunk (about.[hash].js) for this route
              // which is lazy-loaded when the route is visited.
              component: Library,
            },
            {
              path: '/',
              name: 'home',
            }
      ],
    })

    const wrapper = mount(App, { localVue, router })

    router.push('bookmarks')
    expect(wrapper.find('aside').exists()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Tagpanel exists in Library', () => {

    const wrapper = mount(Library, {
    })
    expect(wrapper.find('aside').exists()).toBeTruthy()
  })

  it('BookMarkPanel exits in Library', () => {
    const wrapper = mount(Library, {
    })
    expect(wrapper.find('main').exists()).toBeTruthy()

  })


})
