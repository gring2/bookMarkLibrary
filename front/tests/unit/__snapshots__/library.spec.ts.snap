// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library.vue renders a component via routing 1`] = `
<div
  id="app"
>
  <nav>
    <div>
      <div
        class="home-logo"
      >
        <a
          class="link-style-none router-link-exact-active router-link-active"
          href="/"
        >
          <h1>
            BookMarkLibrary
          </h1>
        </a>
      </div>
       
      <div>
        <ul
          class="list-style-none flex flex-items-center"
        >
          <li>
            <a
              class="manu"
              href="/bookmarks"
            >
              library
            </a>
          </li>
        </ul>
      </div>
    </div>
     
    <div
      class="authenticate flex-items-center"
    >
      <div>
        <p>
          Hi! test email
        </p>
      </div>
       
      <div
        class="logout-box"
      >
        <button
          class="logout"
        >
          Log out
        </button>
      </div>
       
      <div>
        <button
          class="register"
        >
          Sign Up
        </button>
      </div>
       
      <div>
        <button
          class="login"
        >
          Sign In
        </button>
      </div>
    </div>
  </nav>
   
  <section
    class="content"
  >
    <div>
      <aside
        class="side"
      >
        <ul>
          <span>
            test1
          </span>
        </ul>
        <ul>
          <span>
            test2
          </span>
        </ul>
      </aside>
       
      <main>
        <div>
          <input
            placeholder="url"
            style="width: 100%; font-size: 2rem; height: 2rem;"
            type="text"
          />
           
          <input
            placeholder="#tag"
            style="width: 100%; font-size: 2rem; height: 2rem;"
            type="text"
          />
           
          <button
            class="submit"
            style="font-weight: normal; font-size: 1.4rem; width: 8rem;"
          >
            Susbmit
          </button>
        </div>
         
        <div
          class="container"
        >
          <ul>
            <li>
              <div>
                <img
                  alt=""
                  src="https://cdn.qiita.com/assets/qiita-fb-2887e7b4aad86fd8c25cea84846f2236.png"
                />
                 
                <a
                  href="#"
                >
                  title
                </a>
              </div>
            </li>
          </ul>
        </div>
      </main>
    </div>
  </section>
</div>
`;
